bofire/strategies/predictives/botorch.py

```diff
@@ -170,13 +170,21 @@ def _predict(self, transformed: pd.DataFrame) -> Tuple[np.ndarray, np.ndarray]:
         # input and further transform it to a torch tensor
         X = torch.from_numpy(transformed.values).to(**tkwargs)
         with torch.no_grad():
-            posterior = self.model.posterior(X=X, observation_noise=True)  # type: ignore
+            # observation noise is not implemented for MultiTaskGPSurrogate, has to be treated differently
+            if self.surrogate_specs.surrogates[0].type == "MultiTaskGPSurrogate":
```

Comment:
Can you add a validator to `BotorchSurrogates` that checks that in case one `MultiTaskGP` is present, all surrogates are of type `MultiTaskGP`. Else, we will get trouble with the different required encodings. You have to add it here: https://github.com/experimental-design/bofire/blob/main/bofire/data_models/surrogates/botorch_surrogates.py

Reply:
Oh, also important to mention that when I tested with a single surrogate model, or two of them, in all cases it goes through `len(posterior.mean.shape) == 2` which caused my confusion...